File: app/src/main/java/com/android_cademy/starwarstinder/db/ProfileDao.java
package com.android_cademy.starwarstinder.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.android_cademy.starwarstinder.model.Profile;

import java.util.List;

@Dao
public interface ProfileDao {

    @Query("SELECT * FROM Profile")
    LiveData<List<Profile>> getProfiles();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void bulkInsert(Profile... profile);

    @Query("DELETE FROM Profile")
    void deleteProfiles();

    @Query("SELECT COUNT(NAME) FROM Profile")
    int countProfiles();
}

File: app/src/main/java/com/android_cademy/starwarstinder/db/ProfileDao.kt
package com.android_cademy.starwarstinder.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import com.android_cademy.starwarstinder.model.Profile

@Dao
interface ProfileDao {

    @Query("SELECT * FROM Profile")
    fun getAll(): LiveData<List<Profile>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun bulkInsert(vararg profile: Profile)

    @Query("DELETE FROM Profile")
    fun deleteProfiles()

    @Query("SELECT COUNT(NAME) FROM Profile")
    fun countProfiles(): Int
}
